server/client.go

```diff
@@ -3587,7 +3590,11 @@ func removeHeaderIfPresent(hdr []byte, key string) []byte {
 	if end < 0 {
 		return hdr
 	}
-	return append(hdr[:start], hdr[start+end+len(_CRLF_):]...)
+	nhdr := append(hdr[:start], hdr[start+end+len(_CRLF_):]...)
```

Comment:
Few things:
- if the hdr was just the empty line (meaning "NATS/1.0\r\n\r\n"), then this function will return that (since the key is not found). However, if we find the key, and it was the only one, then we return nil. That is not consistent.
- Again, as mentioned previously, you should be careful of not removing a key that is not an exact match, that is, key "Test" would remove the key "Testx", which is bad. Now, this is likely ok since we are looking for a server "key", but since we don't enforce usage of header keys from users, at least let's be strict here.

Reply:
if we do not find anything we should not take any action, regardless of what was passed in. 
I could pass it back after removal and have upper layers null it out if needed.

For making sure its the right key will update.

Reply:
append the separator to key? so we search for "key=" instead of "key"

This still won't help against a key= being some other key's value.

Reply:
ok please take another look, kept the auto return nil if we remove and this was only header present.

Reply:
The suffix is ':' which we now check, but agree could have another prefix. Let me try one more time.